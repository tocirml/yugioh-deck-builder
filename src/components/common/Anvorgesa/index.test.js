import React from 'react';
import { Anvorgesa } from './index';

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';

import { shallow } from 'enzyme';

describe('rendering', () => {
  it('renders without crashing', () => {
    shallow(<Anvorgesa />);
  });

  it('renders 3 bars', () => {
    const wrapper = shallow(<Anvorgesa />);
    expect(wrapper.find('span').length).toEqual(3);
  });
});

describe('toggle', () => {
  it('calls toggleHandler on click', () => {
    const handler = jest.fn();
    render(<Anvorgesa toggleHandler={handler} />);

    userEvent.click(screen.getByTestId('anvorgesa'));

    expect(handler).toHaveBeenCalledTimes(1);
  });
});

describe('snapshots', () => {
  it('Anvorgesa snapshot', () => {
    const tree = shallow(<Anvorgesa />);
    expect(tree).toMatchSnapshot();
  });
});
