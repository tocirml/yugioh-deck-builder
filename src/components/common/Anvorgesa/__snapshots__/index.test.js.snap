// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Anvorgesa snapshot 1`] = `
<a
  className="anvorgesa"
  data-testid="anvorgesa"
  href="#"
  onClick={[Function]}
>
  <span
    className="bar"
  />
  <span
    className="bar"
  />
  <span
    className="bar"
  />
</a>
`;
